.github/workflows/nubots.yaml

```diff
@@ -0,0 +1,179 @@
+# Continuous Integration tests
+name: NUbots CI
+
+# Controls when the action will run.
+on:
+  # Triggers the workflow on push or pull request events but only for the main branch
+  push:
+    branches: [main]
```

Comment:
Do we need this given we cant even push to main?

Reply:
My assumption was that it does a build after PRs are merged. 
They do it in the docs [here](https://docs.github.com/en/actions/reference/workflow-syntax-for-github-actions#on) so I think that's correct??

Reply:
That doesn't answer the question of main being protected?

Reply:
Main is pushed to when PRs are merged to main, is my understanding. The protection just means that that's the only time main is pushed to

Reply:
This runs whenever main is updated, A push to main happens when you merge a PR into main